x-pack/legacy/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -91,25 +91,20 @@ export class WorkspaceHandler {
       await this.checkCommit(bareRepo, targetCommit);
       revision = defaultBranch;
     }
-    let workspaceRepo: Repository;
+    let workspace;
     if (await this.workspaceExists(bareRepo, repositoryUri, revision)) {
-      workspaceRepo = await this.updateWorkspace(repositoryUri, revision, targetCommit);
+      workspace = await this.updateWorkspace(repositoryUri, revision, targetCommit);
     } else {
-      workspaceRepo = await this.cloneWorkspace(bareRepo, repositoryUri, revision, targetCommit);
+      workspace = await this.cloneWorkspace(bareRepo, repositoryUri, revision, targetCommit);
     }
+    const workspaceHeadCommit = workspace.workspaceHeadCommit;
+    const workspaceDir = workspace.workspaceDir;
 
-    const workspaceHeadCommit = await workspaceRepo.getHeadCommit();
```

Comment:
@spacedragon we think that the code block from line 101 to line 12 is redundant, because it's quite similar to `updateWorkspace`.

Reply:
Ok，this code was useful before we change to use worktree.